src/main/java/frc/robot/subsystems/Maflipulator.java

```diff
@@ -0,0 +1,193 @@
+
+    public Maflipulator(final RobotMap.MaflipulatorMap map) { // NOPMD
+        super();
+        flipMotor = map.getFlipMotor();
+        anglePot = map.getMaflipulatorPot();
+
+        anglePID = new PIDController(.01, .0009, 0.0, 0.0, anglePot, (double value) -> {
+            angleCorrection = value;
+        });
+
+        if (anglePot.get() < 180)
+            currentPosition = MaflipulatorSide.kFront;
+        else
+            currentPosition = MaflipulatorSide.kBack;
+
+    }
+
+    @Override
+    public void initDefaultCommand() {
+        // Set the default command for a subsystem here.
+        setDefaultCommand(restrictRotate());
+    }
+
+    public Command manualFlip() {
+        // The command is named "Manual Flip" and requires this subsystem.
+        return new Command("Manual Flip", this) {
+            @Override
+            protected void initialize() {
+                // Called just before this Command runs the first time
+            }
+
+            @Override
+            protected void execute() {
+                flipMotor.set(Robot.xBoxCoPilot.getY(Hand.kRight));
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    protected double restrict(double flipSpeed) {
+        if (currentPosition == MaflipulatorSide.kFront) {
+            if (flipSpeed > 0 && anglePot.get() >= FRONT_UPPER_ANGLE) {
+                flipSpeed = 0;
+            }
+            if (flipSpeed < 0 && anglePot.get() <= FRONT_LOWER_ANGLE) {
+                flipSpeed = 0;
+            }
+        } else {
+            if (flipSpeed > 0 && anglePot.get() <= BACK_UPPER_ANGLE) {
+                flipSpeed = 0;
+            }
+            if (flipSpeed < 0 && anglePot.get() >= BACK_LOWER_ANGLE) {
+                flipSpeed = 0;
+            }
+        }
+        return flipSpeed;
+    }
+
+    public Command restrictRotate() {
+        // The command is named "Restrict Rotate" and requires this subsystem.
+        return new Command("Restrict Rotate", this) {
+
+            @Override
+            protected void execute() {
+                double flipSpeed = Robot.xBoxCoPilot.getY(Hand.kRight);
+                flipSpeed = restrict(flipSpeed);
+                flipMotor.set(flipSpeed);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+
+        };
+    }
+
+    public Command Flip() {
+        return new InstantCommand("Flip", this, () -> {
+
+            Command moveCommand;
+            if (currentPosition == MaflipulatorSide.kFront) {
+                moveCommand = moveToPosition(270);
+                currentPosition = MaflipulatorSide.kBack;
+            } else {
+                moveCommand = moveToPosition(90);
+                currentPosition = MaflipulatorSide.kFront;
+            }
+
+            Scheduler.getInstance().add(moveCommand);
```

Comment:
I recommend using `moveCommand.start()` instead - it does some additional checks